app/models/manageiq/providers/cloud_manager/auth_key_pair.rb

```diff
@@ -33,18 +33,13 @@ def self.create_key_pair_queue(userid, ext_management_system, options = {})
 
   def self.create_key_pair(ems_id, options)
     raise ArgumentError, _("ems cannot be nil") if ems_id.nil?
+
     ext_management_system = ExtManagementSystem.find(ems_id)
```

Comment:
```suggestion
    ext_management_system = resource_search(ems_id, :providers)
```

while you are at it... (sorry)

Reply:
This is core not API so `resource_search` doesn't exist here (`NoMethodError (undefined method 'resource_search' for #<Class:0x00005561c298d248>)`).  Once it is put on the queue we don't have the user context and all rbac has to have already been done.